setup.cfg

```diff
@@ -88,7 +87,7 @@ keywords =
   Python3
 
 [options]
-python_requires = >= 3.6
```

Comment:
Could you keep the compatibility until the next release, at least? aiohttp v3.8 uses proxy.py in tests and still runs the tests against Python 3.6 on that branch. We won't be able to use proxy.py with this change. And the previous release still causes a critical dependency tree problem. Or maybe you could release the code before this PR with py36 and then drop it?

Reply:
Totally.  Won't submit this immediately.  I fear something somewhere will break and make folks unhappy.

Good idea.  Let's make a release and then push this in.

Btw, about release, I earlier restricted `PYPI_TOKEN` for master branch.  Because I wanted to keep the same workflow i.e. release should happen from master branch.  I haven't yet looked deeply into `pypi-release` workflow steps.  Might need to address if they are dependent on `develop` somehow.

So here is what I'll do:

1) PR `v2.4.0` from develop to master
2) Tag `v2.4.0` pointing to master
3) Our PyPi release trigger.

One idea can be, we make a `rc` release first, or may be `a`, `b` releases.  Because this will be the first time we run the release workflow.  Wdyt?

Reply:
Let me also spend more time thinking about it.  I read about a few thing in `asyncio`, but need to try them out.  I feared I might break 3.7+, so was being lazy about it.  But lemme double check if we can maintain 3.6 support.

Reply:
Also note that, usage of `asyncio.run` is in part of the code which is only necessary for `--threaded` execution.  Background, `_run_once` used to be sync, and now it has become async.  Due to which when execution is started in a thread, `run` loop is unable to call async `_run_once`, for which now we use `asyncio.run`.

This possibly could be resolved by starting a new loop for every thread.  PS:  Threaded execution performance is already going down due to unnecessary async overhead.  Adding it's own loop for every request handling in a thread, will likely bring down the performance further down.  When I say down, I mean comparable to the `--threadless` execution.

I was thinking of sharing the event loop with threads, but I read it's unsafe and can lead to unintended behavior.  Still I think we can resolve it *(possibly)*.  Will dig into it over this week.

Reply:
I suppose you could replace `asyncio.run()` with `loop.run_until_complete()`. As for starting many loops, it's most definitely is a bad idea. FWIW, if it's hard for you to bring back 3.6, let's just release the current latest commit on `develop` as 2.4.0 (well, or its beta/rc maybe).

Reply:
> Btw, about release, I earlier restricted `PYPI_TOKEN` for master branch. Because I wanted to keep the same workflow i.e. release should happen from master branch. I haven't yet looked deeply into `pypi-release` workflow steps. Might need to address if they are dependent on `develop` somehow.

I don't think you need to restrict it to a specific branch. The idea is that you can point at any commit sha when triggering the workflow, if that's what you want.

> So here is what I'll do:
> 
> 1. PR `v2.4.0` from develop to master
> 2. Tag `v2.4.0` pointing to master
> 3. Our PyPi release trigger.

That's incorrect. The idea is that you shouldn't have to do multiple steps manually. Because of the human factor, you may make mistakes in a complex multi-step process.
Instead, it's currently set up to be a single-step trigger with approvals along the way. You only need to trigger the workflow and the tagging should happen automatically, after the publishing to the PyPI.

> One idea can be, we make a `rc` release first, or may be `a`, `b` releases. Because this will be the first time we run the release workflow. Wdyt?

Alright. Here's what we can do. Just go to https://github.com/abhinavsingh/proxy.py/actions/workflows/test-library.yml, trigger it, only inputting `2.4.0a0` in the `release-version` field. Wait until it gets to the normal PyPI job, you'll see an approval request on the workflow run page and maybe in your inbox. Confirm it.
Don't worry about releasing to regular PyPI. Since it's an alpha version, people will not upgrade to it by default + you'll be able to `yank` the release on PyPI right away (if you want, that is).
Do this only to confirm that the release pipeline is set up properly. No other reason. If this works, we'll proceed with trying to merge dev to master, and then trying to release its last commit with `2.4.0a1` but explicitly filling out the `release-commitish` field too. Once you're satisfied with the results, we can think about `2.4.0rc0` or going straight to `2.4.0`. But that last decision is to be made later, don't worry. Just do the above to be able to practice the process _in a safe way_.